spacy/schemas.py

```diff
@@ -0,0 +1,188 @@
+from typing import Dict, List, Union, Optional
+from enum import Enum
+from pydantic import BaseModel, Field, ValidationError, validator
+from pydantic import StrictStr, StrictInt, StrictFloat, StrictBool
+from collections import defaultdict
+
+from .attrs import NAMES
+
+
+def validate(schema, obj):
+    """Validate data against a given pydantic schema.
+
+    obj (dict): JSON-serializable data to validate.
+    validator (pydantic.BaseModel): The schema to validate against.
+    RETURNS (list): A list of error messages, if available.
+    """
+    try:
+        schema(**obj)
+        return []
+    except ValidationError as e:
+        errors = e.errors()
+        data = defaultdict(list)
+        for error in errors:
+            err_loc = " -> ".join([str(p) for p in error.get("loc", [])])
+            data[err_loc].append(error.get("msg"))
+        return [f"[{loc}] {', '.join(msg)}" for loc, msg in data.items()]
+
+
+# Matcher token patterns
+
+
+def validate_token_pattern(obj):
```

Comment:
Quick note on the conversion part below: the problem here is that patterns may use symbols (ints) instead of the string values, e.g. `{ORTH: "foo"}` instead of `{"ORTH": "foo"}`. This trips up the validation, because the object seems to be spread into the schema as keyword arguments and you end up with non-string keyword argument names, which causes an error.